pkg/chunk/chunk_store.go

```diff
@@ -408,3 +493,86 @@ func (c *store) convertChunkIDsToChunks(ctx context.Context, chunkIDs []string)
 
 	return chunkSet, nil
 }
+
+type indexWriteCache struct {
+	cache.Cache
+}
+
+func (c *indexWriteCache) Store(ctx context.Context, entries []IndexEntry) {
+	indexEntriesCached.Add(float64(len(entries)))
+
+	keys := make([]string, 0, len(entries))
+	bufs := make([][]byte, 0, len(entries))
+
+	for _, entry := range entries {
+		key := dedupeKey(entry)
+		out, err := proto.Marshal(&entry)
```

Comment:
Why do we write the index entry to the cache?  Isn't the key enough to check equality?

Reply:
We moved to memcache so we now need to hash the keys as they might contain non `utf-8` chars which doesn't work with memcache.

Now that we have hashes, we need to store the actual entry to dedupe.